presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/type/DecodedTimestamp.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.type;
+
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class DecodedTimestamp
```

Comment:
This class is similar to `LongTimestamp`, except it is limited to nanoseconds. What is the advantage to introducing this new class?

Reply:
It's intended to capture the timestamps decoded from various Hive sources (Parquet, ORC etc) in a canonical format, from which they can be converted to a Presto timestamp representation.  They all seem to be using nanosecond granularity.